src/widgets/smart-home/entity-state/entity-state.widget.tsx

```diff
@@ -0,0 +1,94 @@
+import { Center, Loader, Stack, Text, Tooltip } from '@mantine/core';
+import { IconAlertHexagon, IconBraces, IconExclamationMark } from '@tabler/icons-react';
+import { useTranslation } from 'react-i18next';
+import { useConfigContext } from '~/config/provider';
+import { api } from '~/utils/api';
+import { defineWidget } from '~/widgets/helper';
+import { IWidget } from '~/widgets/widgets';
+
+const definition = defineWidget({
+  id: 'smart-home/entity-state',
+  icon: IconBraces,
+  options: {
+    entityId: {
+      type: 'text',
+      defaultValue: 'sun.sun',
+      info: true
+    },
+    displayName: {
+      type: 'text',
+      defaultValue: 'Sun'
+    },
+  },
+  gridstack: {
+    minWidth: 1,
+    minHeight: 1,
+    maxWidth: 12,
+    maxHeight: 12,
+  },
+  component: EntityStateTile,
+});
+
+export type ISmartHomeEntityStateWidget = IWidget<(typeof definition)['id'], typeof definition>;
+
+interface SmartHomeEntityStateWidgetProps {
+  widget: ISmartHomeEntityStateWidget;
+}
+
+function EntityStateTile({ widget }: SmartHomeEntityStateWidgetProps) {
+  const { t } = useTranslation('modules/smart-home/entity-state');
+  const { name: configName } = useConfigContext();
+
+  const { data, isInitialLoading, isLoading, isError, error } =
+    api.smartHomeEntityState.retrieveStatus.useQuery(
+      {
+        configName: configName!,
+        entityId: widget.properties.entityId,
+      },
+      {
+        enabled: !!configName,
+      }
+    );
+
+  const Data = () => {
```

Comment:
Do not define a component within a component. You can rather set the display to the jsx. So for example 
```tsx
let data = null;
if (isError) {
  data = <Tooltip label={error.message} withArrow withinPortal>
          <IconAlertHexagon color="red" />
        </Tooltip>;
}

and so on
```

Reply:
Actually I was thinking about this we should create a component that takes in a condition and displays a widget if that condition is true, otherwise display another widget. It would be really nice in order to show some error states inside of a widget (inside the title/main content)